onnxruntime/python/tools/transformers/bert_perf_test.py

```diff
@@ -119,6 +121,55 @@ def create_session(model_path, use_gpu, provider, intra_op_num_threads, graph_op
 
     return session
 
+def numpy_type(torch_type):
+    type_map = {torch.float32: np.float32,
+                torch.float16: np.float16,
+                torch.int32: np.int32,
+                torch.int64: np.longlong}
+    return type_map[torch_type]
+
+def create_input_output_tensors(inputs, outputs, device):
+    input_tensors = {name: torch.from_numpy(array).to(device)
+                        for name, array in inputs.items()}
+    output_tensors = {name: torch.from_numpy(array).to(device)
+                        for name, array in outputs.items()}
+    return input_tensors, output_tensors
+
+def create_io_binding(sess, input_tensors, output_tensors):
+    io_binding = sess.io_binding()
+    for name, tensor in input_tensors.items():
+        io_binding.bind_input(name, tensor.device.type, 0,
+                                numpy_type(tensor.dtype), tensor.shape,
+                                tensor.data_ptr())
+    for name, tensor in output_tensors.items():
+        io_binding.bind_output(name, tensor.device.type, 0,
+                                numpy_type(tensor.dtype), tensor.shape,
+                                tensor.data_ptr())
+    return io_binding
+
+def onnxruntime_inference_with_io_binding(session, all_inputs, output_names, test_setting):
+    results = []
+    latency_list = []
+    device = 'cuda' if test_setting.use_gpu else 'cpu'
+    for test_case_id, inputs in enumerate(all_inputs):
+        result = session.run(output_names, inputs)
+        outputs = {}
+        for i in range(len(output_names)):
+            outputs[output_names[i]] = result[i]
+
+        input_tensors, output_tensors = create_input_output_tensors(inputs, outputs, device)
+        io_binding = create_io_binding(session, input_tensors, output_tensors)
+
+        # warm up
+        session.run_with_iobinding(io_binding)
+
+        for i in range(test_setting.test_times):
```

Comment:
Similar to onnxruntime_inference does, can we move the test_setting.test_times loop outside of this function, in run_one_test?

Reply:
sure, will make it comparable, although I think both logic would serve the same purpose